src/hooks/transactions/index.tsx

```diff
@@ -79,15 +84,35 @@ export function TransactionsUpdater() {
   )
   const onReceipt = useCallback(
     ({ chainId, hash, receipt }) => {
+      if (receipt?.status === 0) {
+        onTxFail?.(new Error('Transaction failed'), receipt)
+      } else {
+        onTxSuccess?.(receipt.transactionHash, receipt)
+      }
       updateTxs((txs) => {
         const tx = txs[chainId]?.[hash]
         if (tx) {
           tx.receipt = receipt
         }
       })
     },
-    [updateTxs]
+    [updateTxs, onTxFail, onTxSuccess]
   )
 
+  const oldTxs = useRef({})
+  useEffect(() => {
+    const newTxHashes = Object.keys(pendingTransactions)
+    const oldTxHashes = Object.keys(oldTxs.current)
```

Comment:
update to: `oldTxHashes = new Set(Object.keys(oldTxs.current))` and `oldTxHashes.has(txHash)` below

`.includes` is a loop through the array, while `.has` is [strictly better](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#description).